lms/static/js/learner_dashboard/views/program_details_sidebar_view.js

```diff
@@ -1,47 +1,89 @@
+    ],
+        function(
+            Backbone,
+            $,
+            _,
+            gettext,
+            StringUtils,
+            ProgramProgressView,
+            CertificateView,
+            sidebarTpl
+        ) {
+            return Backbone.View.extend({
+                tpl: _.template(sidebarTpl),
+
+                initialize: function(options) {
+                    this.courseModel = options.courseModel || {};
+                    this.certificateCollection = options.certificateCollection || [];
+                    this.programCertificate = this.getProgramCertificate();
+                    this.render();
+                },
+
+                render: function() {
+                    var data = $.extend({}, this.model.toJSON(), {
+                        programCertificate: this.programCertificate ?
+                                            this.programCertificate.toJSON() : {}
+                    });
+
+                    this.$el.html(this.tpl(data));
+                    this.postRender();
+                },
+
+                postRender: function() {
+                    if (!this.programCertificate) {
+                        this.progressModel = new Backbone.Model({
+                            title: StringUtils.interpolate(
+                                gettext('{type} Progress'),
+                                {type: this.model.get('type')}
+                            ),
+                            label: gettext('Earned Certificates'),
+                            progress: {
+                                completed: this.courseModel.get('completed').length,
+                                in_progress: this.courseModel.get('in_progress').length,
+                                not_started: this.courseModel.get('not_started').length
+                            }
+                        });
+
+                        this.programProgressView = new ProgramProgressView({
+                            el: '.js-program-progress',
+                            model: this.progressModel
+                        });
+                    }
+
+                    if (this.certificateCollection.length) {
+                        this.certificateView = new CertificateView({
+                            el: '.js-course-certificates',
+                            collection: this.certificateCollection,
+                            title: gettext('Earned Certificates')
+                        });
+                    }
+                },
+
+                getProgramCertificate: function() {
+                    var certificate = this.certificateCollection.findWhere({type: 'program'}),
+                        base = '/static/images/programs/program-certificate-';
+
+                    if (certificate) {
+                        certificate.set({
+                            img: base + this.model.get('type').toLowerCase() + '.gif'
```

Comment:
is the type here the type of the course run or program?  
The program type would be Professional Certificate, XSeries, MicroMasters, but the course run types i think are verified, professional, credit, etc...